ui/menu.js

```diff
@@ -297,16 +290,18 @@ return $.widget( "ui.menu", {
 			.each(function() {
 				var menu = $( this ),
 					item = menu.prev(),
-					submenuCaret = $( "<span>" )
-						.addClass( "ui-menu-icon ui-icon " + icon )
-						.data( "ui-menu-submenu-caret", true );
+					submenuCaret = $( "<span>" ).data( "ui-menu-submenu-caret", true );
 
+				that._addClass( submenuCaret, "ui-menu-icon", "ui-icon " + icon );
 				item
 					.attr( "aria-haspopup", "true" )
 					.prepend( submenuCaret );
 				menu.attr( "aria-labelledby", item.attr( "id" ) );
 			});
 
+		this._addClass( submenus.filter( ":not(.ui-menu)" ),
```

Comment:
I think we should avoid calling `filter` with this argument twice. Seems like `newSubmenus` is the right variable for that, which is used below. Seems like its actually badly named for how its used, since "submenus" refers to ".ui-menu" elements, while "newSubmenus" refers to wrappers (".ui-menu-item-wrapper").

Whatever the right solution here, this refresh method needs some work.